src/js/tabs/pid_tuning.js

```diff
@@ -1302,8 +1302,32 @@ TABS.pid_tuning.initialize = function (callback) {
         $('input.feature').on('input change', updateRates);
         $('.pid_tuning').on('input change', updateRates).trigger('input');
 
-        $('.throttle input').on('input change', function () {
-            setTimeout(function () { // let global validation trigger and adjust the values first
+        function redrawThrottleCurve(forced) {
+            if (!forced && !TABS.pid_tuning.checkThrottle()) {
+                return;
+            }
+
+            /* 
+            Quadratic curve formula taken from: 
+                https://stackoverflow.com/a/9195706/176210 
+            */
+
+            function getQBezierValue(t, p1, p2, p3) {
+                var iT = 1 - t;
+                return iT * iT * p1 + 2 * iT * t * p2 + t * t * p3;
+            }
+
+            function getQuadraticCurvePoint(startX, startY, cpX, cpY, endX, endY, position) {
+                return {
+                    x:  getQBezierValue(position, startX, cpX, endX),
+                    y:  getQBezierValue(position, startY, cpY, endY),
+                };
+            }
+
+            /* --- */
+
+            setTimeout(function () { 
```

Comment:
What is the reason we are doing this asynchronously?

Reply:
Hm. It seems that the only reason is my misuse of the famous copy-paste technology. I edited the code so now it works more like in current master where calling `setTimeout()` did make sense. In there it is used to let the default HTML input validations trigger before calling the event handler.

Reply:
Getting the triggers to work the way they should is a plus!